### DISCUSSION TITLE
Shell in every app.

### DISCUSSION THREAD

**Discussion by boromyr** [moudey/Shell#65] [Q&A]
Would it be possible to use shell as context menu even in processes other than explorer.exe? For example in XYplorer or in compression programs?

**Comment by moudey** [comment]
**Shell** in the following update supports any program that load system context menu such as Explorer++, Xyplorer, 7zip, etc.
With an `exclude` option to exclude any process or window

**Comment by boromyr** [comment]
Are you referring to the 1.8 beta version? How can I enable this feature?

**Comment by moudey** [comment]
@boromyr Please test this [build](https://github.com/moudey/Shell/files/10127710/shell-test.zip) 

Comment this line to enable shell with other programs
```
exclude.where = !process.is_explorer
```

**Comment by boromyr** [comment]
i tried in a virtual machine with windows 11 22H2 but it didn't work in 7zip, but in bandizip it does

**Comment by boromyr** [comment]
With shell 1.8.1 commenting `exclude.where = !process.is_explorer` works with 7zip and other apps that don't use custom context menus, but unfortunately it doesn't work with XYplorer, I don't know if it is compatible with other file managers out there, or if is an exclusive problem of XY

**Comment by adambisho** [comment]
Shell looks like an amazing program and I really want to use it but I'm having issues.  I am using the app Q-dir from softwareok.com and can't get the shell context menu to work with it.  I commented out the "exclude.where" line and restarted explorer, but that didn't seem to work.  The shell menu works fine with windows explorer.  Any help with this issue?

**Reply by adambisho** [reply]
I think I figured it out.  With Q-dir you must use SHIFT, RIGHT-CLICK to get to the MS context menu.  It looks like the SHELL menu works when doing this.  I'm not a programmer and just getting started with SHELL, but I'm very excited about customizing it.  Thanks for this great app.
